File: ruby/03-switch-led.rb
require 'rpi_gpio'

# Run `gpio readall` to confirm BCM pin number(s)
pin_in = 18   # button
pin_out = 17  # LED

RPi::GPIO.set_numbering(:bcm)
RPi::GPIO.set_warnings(0)
RPi::GPIO.setup(pin_in, as: :input, pull: :down)
RPi::GPIO.setup(pin_out, as: :output)

def blink(pin)
  puts("Button pressed")
  RPi::GPIO.set_high(pin)
  sleep(0.2)
  RPi::GPIO.set_low(pin)
end

begin
  loop do
    if RPi::GPIO.high?(pin_in) # button input state is "on"
      blink(pin_out)
      sleep(0.2)
    end
  end
rescue Interrupt
  puts 'Done'
ensure
  RPi::GPIO.clean_up
end

File: python/03-switch-led.py
import RPi.GPIO as GPIO
import time

# Run `gpio readall` to confirm pin number(s)
pinIn = 18	# button
pinOut = 17	# LED

GPIO.setmode(GPIO.BCM)
GPIO.setwarnings(False)
GPIO.setup(pinIn, GPIO.IN, pull_up_down = GPIO.PUD_DOWN)
GPIO.setup(pinOut, GPIO.OUT)

def blink(channel):
  print("Button pressed")
  GPIO.output(pinOut,GPIO.HIGH)
  time.sleep(0.2)
  GPIO.output(pinOut,GPIO.LOW)

# Call blink when button state rises from "0" to "1"
GPIO.add_event_detect(pinIn, GPIO.RISING, callback=blink, bouncetime=300)

try:
  while True:
    time.sleep(1)
except KeyboardInterrupt:
  print("\nDone")
finally:
  GPIO.cleanup()